### PR TITLE
Simplify CMake build flags

### PR DESCRIPTION
**Pull request by Chocobo1**
* build: Specify LANGUAGES in build script
* build: Use CMake variable to set MSVC runtime library
  https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html#variable:CMAKE_MSVC_RUNTIME_LIBRARY
* build: Use CMake variable for enabling LTCG
* build: Drop redundant macro define
  CMake will set `NDEBUG` automatically:
  https://github.com/Kitware/CMake/blob/50252733c46025d7086d7b0ea6daef04e0c5346f/Modules/Platform/Windows-MSVC.cmake#L488-L491
  MSVC will set `_DEBUG` automatically:
  https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170#microsoft-specific-predefined-macros
* build: Drop compiler flag override files
  Those flags are mostly the same as the ones used by CMake. And for those flags that are
  different, extract them out.
  * https://github.com/Kitware/CMake/blob/50252733c46025d7086d7b0ea6daef04e0c5346f/Modules/Platform/Windows-MSVC.cmake#L488-L491
  * https://learn.microsoft.com/en-us/cpp/build/reference/opt-optimizations?view=msvc-170
* build: Reorder override flag
  The new flag should come after in order to override previous flag.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,6 +10,10 @@ add_definitions(
 	/D_CRT_SECURE_NO_WARNINGS # disable warnings about old libc functions
 )
 set(CMAKE_CXX_FLAGS "/GR- ${CMAKE_CXX_FLAGS}") # turn off C++ RTTI
+set(CMAKE_C_FLAGS_RELEASE "/GL /Gw")
+set(CMAKE_CXX_FLAGS_RELEASE "/GL /Gw")
```
Comment (kanru) [inline CMakeLists.txt:14]: The original flag override adds `/MT` so the redistributable runtime is statically linked to the `.dll`.

Should we keep `/MT` or you want to package the redistributable in the installer instead?

File: CMakeLists.txt [hunk 2]
```diff
@@ -14,6 +10,10 @@ add_definitions(
...
+set(CMAKE_EXE_LINKER_FLAGS_RELEASE "/OPT:REF /OPT:ICF")
```
Comment (kanru) [inline CMakeLists.txt:15]: It looks like `/OPT:REF` and `/OPT:ICF` is already the default linker flag.

https://learn.microsoft.com/en-us/cpp/build/reference/opt-optimizations?view=msvc-170

File: CMakeLists.txt (hunk 1 above)
Comment (Chocobo1) [reply CMakeLists.txt]: oops, my mistake here.
I'll keep the static runtime library via a cmake variable.

File: CMakeLists.txt (hunk 2 above)
Comment (Chocobo1) [reply CMakeLists.txt]: OK, removed.

**Review by kanru — APPROVED** [review]

**Pull request merged**